src/dparse/lexer.d

```diff
@@ -1537,6 +1537,23 @@ private pure nothrow @safe:
             index);
     }
 
+    bool lexNamedEntity()
+    in { assert (range.bytes[range.index] == '&'); }
+    do
+    {
+        Token t;
+        range.popFront();
+        lexIdentifier(t);
+        if (t.type != tok!"identifier" ||
+            (!range.empty && range.bytes[range.index] != ';'))
+        {
+            error("Error: invalid named character entity");
```

Comment:
There's no test for this, is there ?

Reply:
Extended the test to check the actual messages.